### DISCUSSION TITLE
Can't seem to download reblogs or posts from user that requires login

### DISCUSSION THREAD

**Discussion by Lumerica** [TumblThreeApp/TumblThree#313] [Q&A (Question & Answer)]
Hello, 

I am having an issue with the app to be specific with tumblr blogs that requires login and that redirects the link to a different tumblr url.

What I want is try to download things from:
https://www.tumblr.com/blog/view/random/tagged/random
So i was hoping this app to be able to download via https://random.tumblr.com
However the actual thing is that if you open the site in tumblr it will redirect you to: "https://www.tumblr.com/blog/view/random". Which the app does not accept as a valid site to be added I think this is the cause even though I have already authenticated in the app.

-
The situation is:
Upon authenticating in the app, and add the link in the app (https://random.tumblr.com/)
I expect the program to be able download all. this tumblr site requires user login; 

But the app does not download anything of these, only some other posts that I was not planning to.
What did you expect to happen when you encountered this issue?

**Steps to Reproduce the Problem**

Detail the steps below to help us reproduce the problem you encountered.

  1. Go to 'Settings, login tumblr to authenticate and accept privacy terms....'
  2. copied and pasted the original url in the field and clicked on add blog. 
  3. but it only downloads some of the posts. 
  4. Adding the url "https://www.tumblr.com/blog/view/random" is nto possible in the app so i think the problem can be fixed if the app also sees sites with format "https://www.tumblr.com/blog/view/<blogname>" as valid.

**Specifications**
2.5.4.278, windows 10, hardware not relevant

**Comment by thomas694** [comment]
Hello,
thanks for reporting an issue. We'll look into it.

**Comment by thomas694** [comment]
This blog is a 'hidden' blog, so it requires you to log in. Don't get confused how the urls change and are shown in the browser, TumblThree downloads differently than a browser. After adding the blog with https://kpopsteve.tumblr.com/ (or kpopsteve.tumblr.com in short), the blog type is shown as ` Tumblr Hidden` and the crawler type is `SVC` for that type of blogs.

The blog is downloading here without a problem. How many posts do you expect?

> But the app does not download anything of these, only some other posts that I was not planning to.

What kind of things is downloaded instead?

> https://www.tumblr.com/blog/view/kpopsteve/tagged/kara

If you want to only download posts that are tagged, you have to add the blog normally and fill the wanted tags in the tag field.
Seems to work too.

**Comment by Lumerica** [comment]
Hello, thanks a lot for your reply.

I was downloading with the crawler type SVC.

The blog does download, but it seems incomplete after checking.

Actually I left the tag field empty as I wanted to download all and check manually myself. However I see that none of the posts with the tag "kara" are downloaded.

I haven't really counted how many posts there are; i just see that there are many files skipped. You can also see that most of the json files there are without any picture or any other type of files in this directory list:

[listkpopsteve_dl_dir.txt](https://github.com/TumblThreeApp/TumblThree/files/8157302/listkpopsteve_dl_dir.txt)
 
You mentioned the option "check for file existence globally" in my other report, that option I never enabled. Although I did download similar blogs before. But I have yet to check them what is missing; its just that this blog caught my eyes that it downloaded nothing with that tag I mentioned earlier even though I left the field for tags blank it should download all.

**Comment by thomas694** [comment]
After your reply and looking on the folders again, I finally saw it. The newest "kara" image (tagged download) is from mid 2013 and the oldest media from the normal download is from mid 2014. So, you're right, these images aren't in the normal download.

Thinking about that gap, there must be an error that interrupts the download and prevents any older (existing) media from being downloaded. Maybe it's related to a different data/json structure, maybe something else.
I have to check it more in the next days.

Just a note, some of the json files (especially these with numbers only) are without images, because the posts are actually text (and not image) posts, but the app still parses texts for valid image links and downloads them. But the json file isn't renamed or duplicated for every one of these downloaded images.

**Comment by Lumerica** [comment]
No worries, yes I know the numbers only json files are text posts only. Will wait for your reply, thanks a lot for checking this.

**Comment by thomas694** [comment]
It seems to work now. If you like you can download this [test version](https://ci.appveyor.com/api/buildjobs/6srigggki3kar3ge/artifacts/artifacts%2FTumblThree-v2.5.4-x64-Application.zip) and try it out. Otherwise you have to wait until the next official version.

**Comment by Lumerica** [comment]
Hello,

Thank you so much for your help, with the test version it seems to be solved now I got 47k downloads now of the blog mentioned originally.

By the way, if I select SVC crawler mode, should I expect it in general to have better download quality (resolutions etc) results if I use that override option?

Also, you mentioned that the tooltip about the crawler is outdated, is it best to leave the override option unchecked?

**Comment by thomas694** [comment]
Meanwhile you get the same result with both crawlers. So you can uncheck the override option and use the crawlers as they are chosen when adding the blogs.
In the case one crawler is broken and needs an update, for normal blogs the other crawler could be used as a fallback, but hidden blogs only work with the SVC crawler.

**Comment by Lumerica** [comment]
Actually i figured that it's still downloading more with API. so for now I am running first with SVC after its completed wait 4-6 hrs run again with API by just changing the detection method without removing the blog. I think thats most complete way.

**Comment by thomas694** [comment]
The issue has been fixed and closed. You can still comment. Feel free to ask for reopening the issue if needed.
